File: python/14. Longest Common Prefix.py
class Solution(object):
    def longestCommonPrefix(self, strs):
        """
        :type strs: List[str]
        :rtype: str
        """
        if not strs:
            return ''
        pre = strs[0]
        i = 1
        size = len(pre)
        while i < len(strs):
            t = strs[i]
            j = 0
            while j < min(size, len(t)):
                if pre[j] != t[j]:
                    break
                j += 1
            size = min(size, j)
            i += 1
            
        return pre[0:size]

File: go/src/leetcode/14. Longest Common Prefix/14. Longest Common Prefix.go
package main

import (
	"fmt"
	"math"
)

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	t := strs[0]
	for _, str := range strs[1:] {
		j := 0
		for ; j < min(len(t), len(str)); j++ {
			if str[j] != t[j] {
				break
			}
		}
		t = t[:j]
	}
	return t
}

func min(a, b int) int {
	return int(math.Min(float64(a), float64(b)))
}

func main() {
	fmt.Println(longestCommonPrefix([]string{"flower","flow","flight"}))
	//fmt.Println(longestCommonPrefix([]string{"dog","racecar","car"}))
}